src/rendering/vcPolygonModel.cpp

```diff
@@ -245,15 +253,34 @@ udResult vcPolygonModel_CreateFromVSMFInMemory(vcPolygonModel **ppModel, char *p
       if (t == 0)
       {
         void *pTextureData = pFilePos;
-        if (!vcTexture_CreateFromMemory(&pNewModel->pMeshes[i].material.pTexture, pTextureData, textureFileSize))
+        if (!vcTexture_AsyncCreateFromMemory(&pNewModel->pMeshes[i].material.pTexture, pWorkerPool, pTextureData, textureFileSize))
         {
           // TODO: (EVC-570)
         }
       }
       pFilePos += textureFileSize;
     }
 
-    UD_ERROR_CHECK(vcMesh_Create(&pNewModel->pMeshes[i].pMesh, vcP3N3UV2VertexLayout, (int)udLengthOf(vcP3N3UV2VertexLayout), pVerts, pNewModel->pMeshes[i].numVertices, pIndices, pNewModel->pMeshes[i].numElements, vcMF_IndexShort));
+    AsyncPolygonModelLoadMeshInfo* pLoadInfo = udAllocType(AsyncPolygonModelLoadMeshInfo, 1, udAF_Zero);
+    UD_ERROR_NULL(pLoadInfo, udR_MemoryAllocationFailure);
+
+    pLoadInfo->pPolygonModel = pNewModel;
+    pLoadInfo->pMesh = &pNewModel->pMeshes[i];
+    pLoadInfo->pVerts = (vcP3N3UV2Vertex*)udAlloc(vertArraySize);
```

Comment:
`udAllocType`? And below for the indices. Also, you could probably use `udMemdup` instead?

Reply:
Ok